hy/compiler.py

```diff
@@ -1216,20 +1224,69 @@ def _compile_import(expr, module, names=None, importer=asty.Import):
 
         return rimports
 
+    def _compile_subscript(self, root, value, exprs):
+        dims = []
+        ret = Result()
+
+        for expr in exprs:
+            if expr_startswith(HyKeyword(":"), expr):
```

Comment:
Unfortunately, this check isn't strict enough (it allows `(get [1 2 3 4 5] ("\ufdd0:" 1 2))`) because equality of Hy model objects is very loose about type. You probably need to check the type of `expr` explicitly.

Reply:
Would it maybe make sense to have a crack at #1363 and fix that first?

Reply:
You can try, but it's a deep rabbit hole to go down because the compiler currently depends on that "feature" in certain ways. Changes to that may also conflict with #1493. I'd advise waiting for #1493 before you do that, and finishing this first in any case.

Reply:
I can wait, honestly. I'd rather see it some in on a more solid foundation.